lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1714,12 +1718,16 @@ export class UserAgentApplication {
       // Generate and cache accessTokenKey and accessTokenValue
       const accessTokenKey = new AccessTokenKey(authority, this.clientId, scope, clientObj.uid, clientObj.utid);
 
-      // TODO: since there is no access_token, this is also set to id_token?
       const accessTokenValue = new AccessTokenValue(parameters[Constants.idToken], parameters[Constants.idToken], response.idToken.expiration, clientInfo);
       this.cacheStorage.setItem(JSON.stringify(accessTokenKey), JSON.stringify(accessTokenValue));
       accessTokenResponse.scopes = [scope];
       accessTokenResponse.accessToken = parameters[Constants.idToken];
-      accessTokenResponse.expiresIn = response.idToken.expiration;
+      let exp = Number(response.idToken.expiration);
+      if (exp) {
+        accessTokenResponse.expiresIn = new Date(exp * 1000);
```

Comment:
Should this be new Date(Date.now() + (exp * 1000)); or new Date(exp * 1000);?

Reply:
See above comment

Reply:
what is exp if not set ? would the output of this be NaN? 
what if exp is 0 this will be falsey
what is difference between Number(input) vs parseInt(num, 10)

Reply:
Number will either return a number or NaN. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
For empty or null values, Number(x) will return 0. 
For any string values, Number(x) will return NaN. 
NaN evaluates to falsey. https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
Number will parse the entire string, whereas parseInt parses up to the first non-digit and then returns what has been parsed up until then. I think Number is a better fit for this use case.